OrbitGl/CaptureSerializer.cpp

```diff
@@ -190,31 +196,76 @@ ErrorMessageOr<void> CaptureSerializer::Load(std::istream& stream) {
 
   FillFunctionCountMap();
 
-  archive(Capture::GProcessId);
+  Capture::GProcessId = capture_info.process_id();
+  Capture::GProcessName = capture_info.process_name();
 
-  archive(Capture::GProcessName);
+  Capture::GThreadNames = {capture_info.thread_names().begin(),
+                           capture_info.thread_names().end()};
 
-  archive(Capture::GThreadNames);
-
-  archive(Capture::GAddressInfos);
+  Capture::GAddressInfos.clear();
+  Capture::GAddressInfos.reserve(capture_info.address_infos_size());
+  for (const auto& address_info : capture_info.address_infos()) {
+    Capture::GAddressInfos[address_info.absolute_address()] = address_info;
+  }
 
-  archive(Capture::GSamplingProfiler);
   if (Capture::GSamplingProfiler == nullptr) {
     Capture::GSamplingProfiler = std::make_shared<SamplingProfiler>();
   }
-  Capture::GSamplingProfiler->LoadCallstacks();
+  Capture::GSamplingProfiler->ClearCallstacks();
+  for (CallstackInfo callstack : capture_info.callstacks()) {
+    CallStack unique_callstack;
+    unique_callstack.m_Data = {callstack.data().begin(),
+                               callstack.data().end()};
+    Capture::GSamplingProfiler->AddUniqueCallStack(unique_callstack);
+  }
+  for (CallstackEvent callstack_event : capture_info.callstack_events()) {
+    Capture::GSamplingProfiler->AddCallStack(callstack_event);
+  }
   Capture::GSamplingProfiler->ProcessSamples();
 
   time_graph_->Clear();
-
-  archive(*time_graph_->GetStringManager());
+  StringManager* string_manager = time_graph_->GetStringManager();
+  string_manager->Clear();
+  for (const auto& entry : capture_info.key_to_string()) {
+    string_manager->AddIfNotPresent(entry.first, entry.second);
+  }
 
   FillEventBuffer();
+}
+
+ErrorMessageOr<void> CaptureSerializer::Load(std::istream& stream) {
+  google::protobuf::io::IstreamInputStream input_stream(&stream);
+  google::protobuf::io::CodedInputStream coded_input(&input_stream);
+
+  std::string error_message =
+      "Error parsing the capture.\nNote: If the capture "
+      "was taken in a previous Orbit version, it could be incompatible. "
+      "Please check release notes for more information.";
+
+  if (!ReadMessage(&header, &coded_input) || header.version().empty()) {
+    ERROR("%s", error_message);
+    return ErrorMessage(error_message);
+  }
+  if (header.version() != kRequiredCaptureVersion) {
+    std::string incompatible_version_error_message = absl::StrFormat(
+        "This capture format is no longer supported but could be opened with "
+        "%s Orbit version.",
```

Comment:
I would maybe put the version at the end, e.g. "This capture format is no longer supported but could be opened with Orbit version %s."

Reply:
Done